types/src/on_chain_config/consensus_config.rs

```diff
@@ -8,19 +8,163 @@ use move_core_types::account_address::AccountAddress;
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
+pub enum ConsensusAlgorithmConfig {
+    Jolteon {
+        main: ConsensusConfigV1,
+        quorum_store_enabled: bool,
+    },
+    DAG(DagConsensusConfigV1),
+}
+
+impl ConsensusAlgorithmConfig {
+    pub fn default_for_genesis() -> Self {
+        Self::Jolteon {
+            main: ConsensusConfigV1::default(),
+            quorum_store_enabled: true,
+        }
+    }
+
+    pub fn default_if_missing() -> Self {
+        Self::Jolteon {
+            main: ConsensusConfigV1::default(),
+            quorum_store_enabled: false,
+        }
+    }
+
+    pub fn quorum_store_enabled(&self) -> bool {
+        match self {
+            ConsensusAlgorithmConfig::Jolteon {
+                quorum_store_enabled,
+                ..
+            } => *quorum_store_enabled,
+            ConsensusAlgorithmConfig::DAG(_) => false,
+        }
+    }
+
+    pub fn is_dag_enabled(&self) -> bool {
+        match self {
+            ConsensusAlgorithmConfig::Jolteon { .. } => false,
+            ConsensusAlgorithmConfig::DAG(_) => true,
+        }
+    }
+
+    pub fn leader_reputation_exclude_round(&self) -> u64 {
+        match self {
+            ConsensusAlgorithmConfig::Jolteon { main, .. } => main.exclude_round,
+            _ => unimplemented!("method not supported"),
+        }
+    }
+
+    pub fn max_failed_authors_to_store(&self) -> usize {
+        match self {
+            ConsensusAlgorithmConfig::Jolteon { main, .. } => main.max_failed_authors_to_store,
+            _ => unimplemented!("method not supported"),
+        }
+    }
+
+    pub fn proposer_election_type(&self) -> &ProposerElectionType {
+        match self {
+            ConsensusAlgorithmConfig::Jolteon { main, .. } => &main.proposer_election_type,
+            _ => unimplemented!("method not supported"),
+        }
+    }
+
+    pub fn unwrap_dag_config_v1(&self) -> &DagConsensusConfigV1 {
+        match self {
+            ConsensusAlgorithmConfig::DAG(dag) => dag,
+            _ => unreachable!("not a dag config"),
+        }
+    }
+}
+
+const VTXN_CONFIG_PER_BLOCK_LIMIT_TXN_COUNT_DEFAULT: u64 = 1;
```

Comment:
Is 1 too small?

Reply:
It's just a default value. 1 is fine?

Reply:
It sets the value for forge test, etc. But maybe 1 is fine, if jwk will not be enabled on forge?